cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperator.java

```diff
@@ -545,17 +550,45 @@ public void onClose(KubernetesClientException e) {
                                         switch (taskStatus) {
                                             case COMPLETED:
                                                 vertx.cancelTimer(t);
+                                                log.info("Rebalance ({}) is now complete", sessionId);
                                                 p.complete(new KafkaClusterRebalanceStatusBuilder()
                                                         .withSessionId(null)
-                                                        .withOptimizationResult(taskStatusJson.getJsonObject("rebalance").getMap())
+                                                        .withOptimizationResult(taskStatusJson.getJsonObject("summary").getMap())
                                                         .addNewCondition().withType(State.Ready.toString()).endCondition().build());
                                                 break;
                                             case COMPLETED_WITH_ERROR:
-                                                // TODO: Add exception handling and update status?
+                                                // TODO: There doesn't seem to be a way to retrieve the actual error message from the user tasks endpoint?
+                                                //       We may need to propose an upstream PR for this.
+                                                // TODO: We need to add an error field to the Rebalance Status to hold details of any issues while rebalancing
```

Comment:
when there is an error we can just provide the exception for updating the status and it will be used in the status itself. I don't think we need a new error field because the exception information will go into the status condition.

Reply:
Yeah an error in the CO side will propagate to the status but this is talking about an error on the CC side whilst a rebalance (or other operation) is happening and we only find out when the `user_tasks` endpoint is polled. So the issue is that an ongoing rebalance fails and we get `COMPLETED_WITH_ERROR` and we can't tell the user what happened?

Reply:
So is there a sort of "errorMessage" in the returned JSON that you want to print into the status?

Reply:
No, that is the issue. You can't `fetch_original_response` from a `COMPLETED_WITH_ERROR` task so there is no way to tell the user, or for us to react to, what went wrong.

Reply:
so I don't understand what should we print in this new error field in the status :-(

Reply:
Well that `TODO` is really a stage two, after we figure out how to get the error.